### ISSUE TITLE
Support EKS encryption provider

### ISSUE THREAD

**Issue by errm** [terraform-providers/terraform-provider-aws#12291]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS just announced native support for the AWS Encryption Provider in EKS. https://aws.amazon.com/blogs/containers/using-eks-encryption-provider-support-for-defense-in-depth/

I would like for this feature to be supported by the AWS terraform provider!

### New or Affected Resource(s)

* aws_eks_cluster

### Potential Terraform Configuration

```hcl
resource "aws_eks_cluster" "example" {
  # ... other configuration ...
  encryption_config {
    provider {
      key_arn = data.aws_kms_key.example.arn
    }
    resources = ["secrets"]
  }
}
```

🤔 resources could default to `["secrets"]` because *currently it is the only valid value.

**Comment by bflad** [comment]
Hi @errm 👋 Thank you for submitting this. This functionality has already been merged into the Terraform AWS Provider (#12280 / #12287) and will release with version 2.52.0, later today. 👍

**Comment by errm** [comment]
🤦‍♂ I guess you did everything between me reading the blogpost and finishing to write the issue ... 🤷‍♂ thanks!
